File: src/main/java/com/example/transactions/infrastructure/adapters/in/rest/GetStatisticController.java
package com.example.transactions.infrastructure.adapters.in.rest;

import com.example.transactions.application.dto.StatisticResponse;
import com.example.transactions.application.use_cases.GetStatisticService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class GetStatisticController {

    private final GetStatisticService getStatisticService;

    public GetStatisticController(GetStatisticService getStatisticService) {
        this.getStatisticService = getStatisticService;
    }

    @GetMapping(path = "/statistics")
    public ResponseEntity<StatisticResponse> getStatistic() {
        return ResponseEntity.ok(getStatisticService.calculate());
    }
}

File: infrastructure/src/main/kotlin/com/example/infrastructure/in/rest/GetStatisticController.kt
package com.example.infrastructure.`in`.rest

import com.example.application.use_cases.GetStatisticService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class GetStatisticController(private val getStatisticService: GetStatisticService) {

    @GetMapping(path = ["/statistics"])
    fun getStatistic(): ResponseEntity<Any> = ResponseEntity.ok(getStatisticService.calculate())
}
